### DISCUSSION TITLE
[Android]Update FontSize at Runtime programatically in MAUI.

### DISCUSSION THREAD

**Discussion by GnanaPriyaNamasivayam** [dotnet/maui#17895] [Q&A] [answered]
Hi team,

I have a requirement to update font size for label, entry, and 3rd party combobox when the user taps on a button in Shell.TitleView.

Implemented below code to update custom value,

  

```
var currentFontSize = this.label.FontSize;  // Initially we have size as **14 & it was set in Resources->Styles -> styles.xaml.**
  var FontSize15= new StringReader("^label { font-size: 15; }  ^entry { font-size: 15; } ^sfcombobox { font-size: 15");
  var FontSize14 = new StringReader("^label { font-size: 14; }  ^entry { font-size: 14; } ^sfcombobox { font-size: 14");
  if (DeviceInfo.Current.Idiom == DeviceIdiom.Phone)
  {
      if (currentFontSize == 14)
      {
          using (FontSize15)
          {
              this.Resources.Add(StyleSheet.FromReader(FontSize15)); **// On first time when i tab the button it hits & value udpated to 15**
          }
      }
      else if(currentFontSize == 15)
      {
          using (FontSize14)
          {
              this.Resources.Remove("^label { font-size: 15; }  ^entry { font-size: 15; } ^sfcombobox { font-size: 15");

              this.Resources.Add(StyleSheet.FromReader(FontSize14)); **// On second time when i tab the button it hits & value  **NOT** udpated to 14**
          }
      }

  }
```
**Tested in Android, ye to in iOS**

Could anyone help to resolve this issue?

Thanks,
Priya

**Comment by jonmdev** [comment, marked as answer]
Can you explain what the purpose of the lines:

```
              this.Resources.Remove("^label { font-size: 15; }  ^entry { font-size: 15; } ^sfcombobox { font-size: 15");
              this.Resources.Add(StyleSheet.FromReader(FontSize14)); **// On second time when i tab the button it hits & value  **NOT** udpated to 14**
```

Are? I don't use "StyleSheets" so I am not sure how these work. But what is the point? If you just want to set the label.FontSize why not create an event in your app wherever it needs to happen like:

```
public event Action<int> newFontSizeEvent = null;
```
Then on your TapGestureRecognizer or however you're recognizing the user input to change the font size have it run:

```newFontSizeEvent?.Invoke(newFontSizeInt);```

And have your labels that need to respond to this subscribed to the event like:

```Label label = new Label();
newFontSizeEvent += delegate (int fontSize) {
label.FontSize = fontSize;
}
```
Something like that is how I would handle it personally from what you're describing.

**Reply by GnanaPriyaNamasivayam** [reply]
This is how I have added now as a workaround however that meets expectation. 
But in the above query my doubt is how can i change the stylessheets at runtime time. 
Ex. Just consider 3 stylesheets and change it on some event trigger. 

When I tried this, it's changed first time but not subsequent times.
